packages/eslint-plugin/docs/rules/no-unnecessary-condition.md

```diff
@@ -47,6 +47,34 @@ function bar<T>(arg: string) {
 ].filter(t => t); // number[] is always truthy
 ```
 
+:::note
```

Comment:
I suggest moving it out of "examples", because the content is turned into tabs so we want each section to be approximately equal height. My immediate thought is to add it as a bullet under "When not to use it" and explain that if you run into `#9998` you'll likely want to `eslint-disable-next-line`.